src/framework/components/collision/system.js

```diff
@@ -802,12 +803,13 @@ class CollisionComponentSystem extends ComponentSystem {
         if (component && component._hasOffset) {
             const lo = component.data.linearOffset;
             const ao = component.data.angularOffset;
+            const newOrigin = p2;
 
-            quat.copy(rot).transformVector(lo, vec3);
-            vec3.add((pos));
+            quat.copy(rot).transformVector(lo, newOrigin);
+            newOrigin.add(pos);
             quat.copy(rot).mul(ao);
 
-            origin.setValue(vec3.x, vec3.y, vec3.z);
```

Comment:
If it is a relative collision shape (using an offset), both `pos` and `vec3` are referencing the same `pc.Vec3` object